examples/cross-contract-high-level/Cargo.toml

```diff
@@ -16,7 +16,7 @@ near-sdk-sim = { path = "../../near-sdk-sim" }
 [profile.release]
 codegen-units = 1
 # Tell `rustc` to optimize for small code size.
-opt-level = "s"
+opt-level = "z"
```

Comment:
I feel that near-sdk might need more fundamental changes to lower its size. In particular, it depends on serde and serde-json, and those too produce a lot of code bloat (`derive_more` in primitives-core also feels suspicious). https://github.com/not-fl3/nanoserde might be an interesting alternative, although it also relies on monomorphisation and not on the reflecion. Perhaps something like nanoborsh (borshik ?)  could help:


```
11:39:42|~/projects/near-sdk-rs/examples/fungible-token|master⚡?
λ cargo llvm-lines --release | head -n 20
   Compiling fungible-token v0.3.0 (/home/matklad/projects/near-sdk-rs/examples/fungible-token)
    Finished release [optimized] target(s) in 0.53s
  Lines        Copies      Function name
  -----        ------      -------------
  9206 (100%)  372 (100%)  (TOTAL)
   556 (6.0%)    1 (0.3%)  <alloc::vec::Vec<T> as borsh::de::BorshDeserialize>::deserialize
   432 (4.7%)    6 (1.6%)  core::option::Option<T>::map
   293 (3.2%)   38 (10.2%) core::ptr::drop_in_place
   240 (2.6%)    3 (0.8%)  borsh::ser::BorshSerialize::try_to_vec
   230 (2.5%)    2 (0.5%)  borsh::de::BorshDeserialize::try_from_slice
   212 (2.3%)    1 (0.3%)  <fungible_token::Account as borsh::de::BorshDeserialize>::deserialize
   212 (2.3%)    1 (0.3%)  fungible_token::FungibleToken::refund_storage
   193 (2.1%)    1 (0.3%)  fungible_token::FungibleToken::transfer_from
   184 (2.0%)    2 (0.5%)  <near_sdk::collections::lookup_map::LookupMap<K,V> as borsh::de::BorshDeserialize>::deserialize
   177 (1.9%)    1 (0.3%)  borsh::ser::serialize_slice
   174 (1.9%)    4 (1.1%)  alloc::alloc::box_free
   165 (1.8%)    1 (0.3%)  <fungible_token::Account as borsh::ser::BorshSerialize>::serialize
   151 (1.6%)    3 (0.8%)  core::result::Result<T,E>::map_err
   151 (1.6%)    1 (0.3%)  <fungible_token::FungibleToken as borsh::de::BorshDeserialize>::deserialize
   149 (1.6%)    1 (0.3%)  <[T] as borsh::ser::BorshSerialize>::serialize
   144 (1.6%)    2 (0.5%)  core::cmp::max_by
   139 (1.5%)    1 (0.3%)  fungible_token::FungibleTokenContract::transfer_from
```